source/common/router/router.cc

```diff
@@ -1161,6 +1193,13 @@ void Filter::onUpstreamComplete(UpstreamRequest& upstream_request) {
   if (!downstream_end_stream_) {
     upstream_request.resetStream();
   }
+
+  // If gRPC request is finished without grpc-status code, use the one from the
+  // response headers.
+  if (grpc_request_ && http_status_code_ != 0) {
+    upstream_request.upstream_host_->outlierDetector().putHttpResponseCode(http_status_code_);
+  }
```

Comment:
should you set the http_status_code_ for all requests?

Reply:
maybe rename http_status_code- to something like "http_status_from_grpc_response_"

Reply:
Done. Changed the variable name to something which makes more sense.